### PR TITLE
Initial commit of csplib

### PR DESCRIPTION
**Pull request by rgeyer**
CSP-mixin is a jsonnet library of files for creating CSP specific dashboards for a variety of services.

It also includes a complete mixin for generating dashboards of each service type in the library.

This PR is for the first dashboard of object storage (GCP bucket storage, Azure blob storage, AWS S3 in the future.)

Azure Blob storage dashboard:
image
image

Google Blob storage dashboard:
image
image

### CODE REVIEW

**Review by v-zhuravlev — COMMENTED** [review]
Should we change all titles to  Sentence case?

**Comment by v-zhuravlev** [comment]
Is it possible to share sample screenshot?

File: csp-mixin/.lint [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    reason: "Using 'Bucket' as instance label"
+    entries:
+    - dashboard: "Azure Blob Storage"
+    - dashboard: "GCP Blob Storage"
+  panel-datasource-rule:
+    reason: "Many panels use --Mixed-- DS"
+  panel-units-rule:
+    reason: "Unit not necessary or provided in an override"
+    entries:
+    - panel: "API requests by type"
```
Comment (Dasomeone) [inline csp-mixin/.lint:20]: Nit: Missing newline

File: csp-mixin/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# CSPlib
+
+A collection of dashboards and their component parts for cloud service provider SaaS services.
```
Comment (Dasomeone) [inline csp-mixin/README.md:3]: It'd be good to have a sample screenshot here, just to give a quick overview of what's included

File: csp-mixin/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# Notes
+
+## Blob Storage
+* GCP - https://cloud.google.com/monitoring/api/metrics_gcp#gcp-storage
+  * The `quota` metrics are alpha, and don't seem to be getting fetched by alloy, even when enabled as a metrics prefix. Perhaps this needs to be enabled for a project?
```
Comment (Dasomeone) [inline csp-mixin/README.md:9]:
```suggestion
  * The `quota` metrics are alpha, and don't seem to be getting fetched by Grafana Alloy, even when enabled as a metrics prefix. Perhaps this needs to be enabled for a project?
```

Added suggestion for added clarity given this is a public repo

File: csp-mixin/dashboards.libsonnet [hunk 4]
```diff
@@ -0,0 +1,29 @@
+local g = import './g.libsonnet';
+local commonlib = import 'common-lib/common/main.libsonnet';
+{
+  new(csplib):
+    local this = self;
+    {
+      [csplib.config.uid + '-blobstorage.json']:
+        local variables = csplib.signals.blobstore.getVariablesMultiChoice();
+        g.dashboard.new(csplib.config.dashboardNamePrefix + 'Blob Storage')
```
Comment (Dasomeone) [inline csp-mixin/dashboards.libsonnet:9]: To Vitaly's point, style guide says to use sentence case for dashboard titles

**Review by Dasomeone — COMMENTED** [review]
Couple tentative comments, but I find it difficult to review without seeing a screenshot or two of the final product. Bit too easy to lose track with just the code :sweat_smile:

**Comment by rgeyer** [comment]
@v-zhuravlev @Dasomeone Added screenshots, and merged feedback.

*whew* It's like I haven't written a dashboard in a while! Thanks for the feedback, and patience.

File: csp-mixin/rows.libsonnet [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+          this.grafana.panels.apiErrorRate
+          + g.panel.timeSeries.gridPos.withW(12)
+          + g.panel.timeSeries.gridPos.withH(12),
+        ]
+      ),
+      network: [
+        g.panel.row.new('Network'),
+        this.grafana.panels.network
+        + g.panel.timeSeries.gridPos.withW(24)
+        + g.panel.timeSeries.gridPos.withH(12),
```
Comment (Dasomeone) [inline csp-mixin/rows.libsonnet:50]: Similarly for network, 12 height feels like *quite* a lot here. Would expect 6 or 8 for any other integration

**Review by Dasomeone — COMMENTED** [review]
Hey @rgeyer thanks for adding the screenshots! Made it much easier to catch a few things I would've otherwise overlooked for sure.

Couple comments, but aside from some sizing/layout there's only really one panel that I feel needs actual attention :)

File: csp-mixin/panels.libsonnet [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+      + g.panel.timeSeries.fieldConfig.defaults.custom.stacking.withMode('normal'),
+    apiErrorRate:
+      this.signals.blobstore.apiRequestErrorRate.asTimeSeries(),
+    network:
+      commonlib.panels.network.timeSeries.traffic.new('Network traffic', targets=[])
+      + commonlib.panels.network.timeSeries.traffic.withNegateOutPackets()
+      + g.panel.timeSeries.fieldConfig.defaults.custom.stacking.withMode('normal')
+      + this.signals.blobstore.networkRx.asPanelMixin()
+      + this.signals.blobstore.networkTx.asPanelMixin(),
+  },
```
[the remark this answers is not in the document]
Comment (v-zhuravlev) [reply csp-mixin/panels.libsonnet:167]: Overall, I like it the way it is. Moving legend to right + making a table with some aggs could help or converting it from `Network traffic` to `Top 10 network traffic panel` if applicable.

**Review by v-zhuravlev — APPROVED** [review]

**Comment by rgeyer** [comment]
Thanks for the follow-up reviews.

I agree that the timeseries are a *bit* tall, tho as Vitaly pointed out, that's often because the legend would strangle the timeseries.

In the interest of shipping these, I'm going to leave them as-is for now, and wait for end-user feedback. I suspect we'll be iterating on these.

**Pull request merged**